flathub.json

```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["i386"]
```

Comment:
Why not `x86_64` instead and using freedesktop i386 compat extensions just like, for example, Steam?

Reply:
> getting all dependencies (GLEW, GLU) compiled to 32bit is proving to be a very daunting task, and it's not certain that it will even be compatible with current clients and servers.

Also a factor with Steam, is that Steam is a 64bit application with a multiarch dependency structure. ET:Legacy has to be a 32bit application. 64bit support is not even worked unstream because it's considered not worth it.

One of the manual pages even disadvices mixing 32bit applications with 64bit:
https://github.com/flatpak/flatpak/wiki/Multiarch

Reply:
Apart from the amount of work needed to build every single dependency, it shouldn't cause any incompatibilities with existing servers if it works as it is now.

The switch to the compat extension needs to happen at some point. 18.08 will be supported for one more year but that's it.

Reply:
I don't think it's a blocker per se for the time being. I feel it will bite us when 18.08 is EOL though.

Reply:
Ow, is that not very short for 18.08? I was under the impression that this system of Flatpak and runtimes would make it easier to keep older applications compatible for a longer time. Right now, compiling something for every Linux version out there is a hell of a job... I thought that using stand-alone runtimes, it would be easier to keep older applications around. 

I don't want to encourage people to keep their application stuck on 18.08 forever, but for many abandoned applications, that would well be the future. Imagine non-FLOSS applications using Flatpak: Many will be abandoned (or put on maintenance) after a few years, meaning that their runtime version essentially becomes their expiration date. Two years is very short in that context. I don't want win32 scenarios where the core set of libraries has to stay unchanged for 20 years, but some middle ground would be good. Is there a better place to discuss this concern of mine?

Also, in the current context, if 18.08 will only be supported for another year, then I agree with you guys. @barthalion Could you flag the pull request as 'changes required' so that I can look in it later.

Reply:
It's not as if 18.08 is going to disappear anywhere as such. It will stop getting security patches at some point though and desktop integration might stop getting improvements. I would imagine most of the time pushing to newer runtime would at worst mean bundling more support libraries if stuff doesn't get compiled during flatpak build.